common/src/main/java/io/netty/util/concurrent/CompletionStageAdapter.java

```diff
@@ -0,0 +1,527 @@
+/*
+ * Copyright 2019 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.concurrent;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * Wraps a {@link Future} and provides a {@link CompletionStage} implementation on top of it.
+ *
+ * @param <V> the value type.
+ */
+final class CompletionStageAdapter<V> implements CompletionStage<V> {
+    private static final Object MARKER = new Object();
+    private static final Object ERROR_MARKER = new Object();
+
+    private final Future<V> future;
+
+    CompletionStageAdapter(Future<V> future) {
```

Comment:
Nit: Replace constructor with static factory method: `wrap()`, or `adapt()`?

Reply:
Not sure what this would buy us...

Reply:
You're right, not worth it for internal-only.